hoomd/dem/DEM3DForceCompute.cc

```diff
@@ -60,6 +60,66 @@ DEM3DForceCompute<Real, Real4, Potential>::DEM3DForceCompute(
         }
     }
 
+template<typename Real, typename Real4, typename Potential>
+void DEM3DForceCompute<Real, Real4, Potential>::connectDEMGSDShapeSpec(std::shared_ptr<GSDDumpWriter> writer)
+    {
+    typedef hoomd::detail::SharedSignalSlot<int(gsd_handle&)> SlotType;
+    auto func = std::bind(&DEM3DForceCompute<Real, Real4, Potential>::slotWriteDEMGSDShapeSpec, this, std::placeholders::_1);
+    std::shared_ptr<hoomd::detail::SignalSlot> pslot(new SlotType(writer->getWriteSignal(), func));
+    addSlot(pslot);
+    }
+
+template<typename Real, typename Real4, typename Potential>
+int DEM3DForceCompute<Real, Real4, Potential>::slotWriteDEMGSDShapeSpec(gsd_handle& handle) const
+    {
+    GSDShapeSpecWriter shapespec(m_exec_conf);
+    m_exec_conf->msg->notice(10) << "DEM3DForceCompute writing to GSD File to name: " << shapespec.getName() << std::endl;
+    int retval = shapespec.write(handle, this->getTypeShapeMapping(m_shapes, m_evaluator.getRadius()));
+    return retval;
+    }
+
+template<typename Real, typename Real4, typename Potential>
+std::string DEM3DForceCompute<Real, Real4, Potential>::getTypeShape(const std::vector<vec3<Real>> &verts, const Real &radius) const
+    {
+    std::ostringstream shapedef;
+    unsigned int nverts = verts.size();
+    if (nverts == 1)
```

Comment:
How do we handle `nverts == 2`? We don't have a spherocylinder definition. I'm not sure if spherocylinders would work, going from HOOMD to GSD to OVITO.

Reply:
Good point. Perhaps we could just error out for now?

Reply:
Good point. Perhaps we could just error out for now?

Reply:
Just throwing and error for now.